docs/advanced/building-plugins.md

```diff
@@ -1,13 +1,70 @@
 # Building plugins
 
-This section is an overview of how to create a plugin. For a complete example of a
-plugin go to the [TypeScript plugin boilerplate project](https://github.com/nomiclabs/hardhat-ts-plugin-boilerplate/).
+In this section, we will explore the creation of plugins for Hardhat, which are the key component
+for integrating other tools and extending the built-in functionality.
 
-## Plugin functionality
+## What exactly are plugins in Hardhat?
 
-Plugins are bits of reusable configuration. Anything that you can do in a plugin, can also be done in your config file. You can test your ideas in a config file, and move them into a plugin when ready.
+Plugins are bits of reusable configuration. Anything that you can do in a plugin, can
+also be done in your config file. You can test your ideas in a config file, and move
+them into a plugin when ready.
 
-The main things that plugins can do are extending the Hardhat Runtime Environment, extending the Hardhat config, defining new tasks, and overriding existing ones.
+When developing a plugin the main tools available to integrate new functionality are extending the
+[Hardhat Runtime Environment], extending the Hardhat config, defining new tasks and
+overriding existing ones, which are all configuration actions achieved through code.
+
+Some examples of things you could achieve by creating a plugin are running a linter when
+the `check` task runs, using different compiler versions for different files or
+generating an UML diagram for your contracts.
+
+
+## Extending the Hardhat Runtime Environment
+
+Let’s go through the process of creating a plugin to inject ethers.js to the [Hardhat Runtime Environment].
```

Comment:
```suggestion
Let’s go through the process of creating a plugin to inject ethers.js to the Hardhat Runtime Environment.
```

Reply:
This is totally outdated. Will change it now